### ISSUE TITLE
Use vim commands in keybinding with vim mode disabled

### ISSUE THREAD

**Issue by Spoutnik97** [zed-industries/zed#22191]
### Check for existing issues

- [X] Completed

### Describe the feature

I would like to use some vim features in normal mode (vim mode disabled). 
For instance I would like to setup a keybinding (e.g. "alt-{" )to simulate the "ci{" vim command to change all the content between "{ }" in my code. 

1. Is it possible?
2. If yes, then how to do this?

### Zed Version and System Specs

0.165.4

**Comment by baldwindavid** [comment]
I don't think it is possible. The closest I can think of is enabling vim mode, but staying in insert mode all the time and using some sort of hack like...

```json
"ctrl-[": ["workspace::SendKeystrokes", "escape c i {"],
```

So while you're in vim's insert mode (what you're calling _normal_ mode above) this keybinding would first flip to vim's normal mode and then enter vim's insert mode again with the characters inside `{` removed.

I don't believe there's a way to start Zed in vim's insert mode per file though so you'd still need to press `i` to get into that for each opened file.

Certainly my real suggestion would be to actually use vim and enter insert mode as little as possible, but I get that's not what you're looking for.

**Comment by notpeter** [comment]
While this is an inserting thought, and I understand the desire to expose the functionality of vim without using vim mode, it's not something which is currently possible without significant architectural changes.
